File: chirp-impl/src/main/java/sample/chirper/chirp/impl/ChirpModule.java
/*
 * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
 */
package sample.chirper.chirp.impl;

import com.google.inject.AbstractModule;
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport;

import sample.chirper.chirp.api.ChirpService;

public class ChirpModule extends AbstractModule implements ServiceGuiceSupport {
  @Override
  protected void configure() {
    bindServices(serviceBinding(ChirpService.class, ChirpServiceImpl.class));
  }
}

File: chirp-impl/src/main/scala/sample/chirper/chirp/impl/ChirpModule.scala
/*
 * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
 */
package sample.chirper.chirp.impl

import com.google.inject.AbstractModule
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport

import sample.chirper.chirp.api.ChirpService

class ChirpModule extends AbstractModule with ServiceGuiceSupport {
  
  protected override def configure(): Unit =
    bindServices(serviceBinding(classOf[ChirpService], classOf[ChirpServiceImpl]))
}
